File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* dummyNode=new ListNode(-1);
        dummyNode->next=head;
        ListNode* curr=dummyNode;
        while(curr->next!=NULL && curr->next->next!=NULL){
            ListNode* swap1=curr->next;
            ListNode* swap2=curr->next->next;
            swap1->next=swap2->next;
            curr->next=swap2;
            swap2->next=swap1;
            curr=swap1;
        }
        return dummyNode->next;
    }
};

File: 24-swap-nodes-in-pairs/swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode | None) -> ListNode | None:
        if head==None or head.next==None:
            return head
        dummy=ListNode()
        dummy.next=head
        curr=dummy
        while curr.next and curr.next.next:
            n1=curr.next
            n2=curr.next.next
            temp=n2.next
            curr.next=n2
            n2.next=n1
            n1.next=temp
            curr=n1
        return dummy.next
            
            